doc/application/application.rst

```diff
@@ -98,8 +109,26 @@ Follow these steps to create a new application directory. (Refer to
       cd app
       mkdir src
 
-#. Create a :file:`CMakeLists.txt` file in your application directory with the
-   following contents:
+#. Create an empty :file:`CMakeLists.txt` file in your application directory.
+
+#. Optionally include additional :file:`.conf` files in your
```

Comment:
I like the idea about adding more information on conf files in a parallel way to DTS overlay files, but:

- there is already a list item about that in this section (see "Create one or more files containing your application's configuration" in this same list), and having two is redundant
- for a first-time reader who lacks context, it might be confusing when this is necessary, and what this `boards/${BOARD}.conf` is for.

How about keeping it simple:

```
#. Configure features used by your application. Usually, you just create a file
   named :file:`prj.conf` in your application directory, where you enable or disable
   features provided by Zephyr's configuration system. You can use existing samples
   to get started. For more information, see :ref:`application_configuration` below.
```

The "per-board mixins" trick you're hinting at here can then go below in the detailed section.

Either way, this item and the one below should be unified into one.